lib/jekyll/site.rb

```diff
@@ -424,7 +424,7 @@ def configure_plugins
     private
     def configure_theme
       self.theme = nil
-      self.theme = Jekyll::Theme.new(config["theme"]) if config["theme"]
+      self.theme = Jekyll::Theme.new(config["theme"]) if config["theme"].is_a?(String)
```

Comment:
```ruby
self.theme = if ...
end
```

Will result in the same thing as those two lines.  That'll rely on you using the if on the setter rather than setting if the condition is met.  However it's a little more optimized that way and results in less messages.